daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-patterns.md

```diff
@@ -586,7 +586,64 @@ The key takeaways from this example are:
 - The number of parallel tasks can be static or dynamic
 - The workflow itself is capable of aggregating the results of parallel executions
 
-While not shown in the example, it's possible to go further and limit the degree of concurrency using simple, language-specific constructs. Furthermore, the execution of the workflow is durable. If a workflow starts 100 parallel task executions and only 40 complete before the process crashes, the workflow restarts itself automatically and only schedules the remaining 60 tasks.
+Furthermore, the execution of the workflow is durable. If a workflow starts 100 parallel task executions and only 40 complete before the process crashes, the workflow restarts itself automatically and only schedules the remaining 60 tasks.
+
+It's possible to go further and limit the degree of concurrency using simple, language-specific constructs.
+
+{{< tabs ".NET" >}}
+
+{{% codetab %}}
+<!-- .NET -->
+```csharp
+public static class TaskExtensions
+{
+  public static async Task<IEnumerable<T>> WhenAllWithLimitAsync<T>(this IEnumerable<Task<T>>> tasks, string activityName, int maxDegreeOfParallelism)
+  {
+    var results = new List<T>();
+    var inFlight = new HashSet<Task<T>>();
+    foreach (var task in tasks)
+    {
+      if (inFlight.Count > maxParallelism)
+      {
+        var finishedTask = await Task.WhenAny(inFlight);
+        results.Add(finishedTask.Result);
+        inFlight.Remove(finishedTask);
+      }
+
+      inFlight.Add(context.CallActivityAsync<int>(task))
+    }
+
+    //Wait for all the remaining tasks to complete
+    await Task.WhenAll(inFlight);
```

Comment:
This function doesn't actually return any results, AFAICT. I think you'd need to change this line to accumulate the results and then add a `return` statement to return those results.
```csharp
results.AddRange(await Tasks.WhenAll(inFlight));
return results;
```

Reply:
Yep, forgot to re-add the Task.WhenAll line.